flink-connectors/flink-connector-kafka-0.10/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/Kafka010Fetcher.java

```diff
@@ -228,4 +226,33 @@ protected void doCommitInternalOffsetsToKafka(
 		// record the work to be committed by the main consumer thread and make sure the consumer notices that
 		consumerThread.setOffsetsToCommit(offsetsToCommit, commitCallback);
 	}
+
+	private class KafkaCollector implements Collector<T> {
+		private final Queue<T> records = new ArrayDeque<>();
+
+		private boolean endOfStreamSignalled = false;
+
+		@Override
+		public void collect(T record) {
+			// do not emit subsequent elements if the end of the stream reached
```

Comment:
So we're not emitting all the records of the batch after one of them signals "end of stream"? I recall that an initial version still emitted all records.

Reply:
Actually, that is the behaviour I had from the very beginning. Do you think it makes more sense to emit all but the end of the stream record?

Side note. I find this method very confusing. I just realized that there is no cross partition alignment on this method. The whole task will be brought down if any of the assigned partitions signals the end of stream, irrespective of the state of the remaining partitions. Honestly I'd be in favour of dropping this method at some point in the future.

Reply:
Oh boy, I guess the interplay of this end record and multiple partitions was not considered... 🙈

Reply:
It had no additional implications on this PR as it has those problems already.